libraries/provider_install_package.rb

```diff
@@ -0,0 +1,65 @@
+class Chef
+  # Chef Provider for installing or removing Elasticsearch from package or source
+  class Provider::ElasticsearchInstallPackage < Provider
+    include Poise
+    include ElasticsearchCookbook::Helpers
+
+    if respond_to?(:provides)
+      provides :elasticsearch_install
+    end
+
+    def action_install
+      converge_by("#{new_resource.name} - install package") do
+        notifying_block do
+
+          package_url = get_package_url(new_resource, node)
+          filename = package_url.split('/').last
+
+          remote_file "#{Chef::Config[:file_cache_path]}/#{filename}" do
+            source package_url
+            checksum get_package_checksum(new_resource, node)
+            mode 00644
+          end
+
+          # this is the only platform-specific thing we're doing in the provider,
+          # so we're not going to break this out into two providers
+          if node['platform_family'] == 'debian'
```

Comment:
What about doing a `case` here instead, where we check for `debian` and `rhel`/`centos`, and raise an exception "Unsupported operating system" or such when it's none of those (e.g. Mac OS X, Solaris, etc)? How would we go about supporting those systems? Write custom provider classes in that case? Is it better to start the separation now or wait?